R/classify-diabetes.R

```diff
@@ -0,0 +1,105 @@
+#' Classify diabetes status using Danish registers.
+#'
+#' @return The same object type as the input data, which would be a
+#'   [tibble::tibble()] type object.
+#' @export
+#' @seealso See the [vignette("function-flow", package = "osdc")] for a detailed
+#'   description of the internal implementation of this classification function.
+#'
+#' @examples
+#' classify_diabetes(
+#'   kontakter = register_data$kontakter,
+#'   diagnoser = register_data$diagnoser,
+#'   lpr_diag = register_data$lpr_diag,
+#'   lpr_adm = register_data$lpr_adm,
+#'   sysi = register_data$sysi,
+#'   sssy = register_data$sssy,
+#'   lab_forsker = register_data$lab_forsker,
+#'   bef = register_data$bef,
+#'   lmdb = register_data$lmdb
+#' )
+classify_diabetes <- function(kontakter, diagnoser, lpr_diag, lpr_adm, sysi, sssy, lab_forsker, bef, lmdb) {
+  # Verification step -----
+  verify_required_variables(kontakter, "kontakter")
+  verify_required_variables(diagnoser, "diagnoser")
+  verify_required_variables(lpr_diag, "lpr_diag")
+  verify_required_variables(lpr_adm, "lpr_adm")
+  verify_required_variables(sysi, "sysi")
+  verify_required_variables(sssy, "sssy")
```

Comment:
Oh, I've included this in `include_podiatrist_services()` too. Should it only be here?